tests/test_lab.py

```diff
@@ -180,10 +180,16 @@ def test_ilab_commands_tested():
 def test_ilab_missing_config(command: Command, cli_runner: CliRunner) -> None:
     cmd = command.get_args(default_config=False)
     assert "--config" not in cmd
+
+    # make sure cli doesn't attempt to read config from the default location
+    # (which may actually exist)
+    if command.needs_config:
+        cmd = ["--config", "./non-existing.yaml"] + cmd
```

Comment:
I put this here because for some reason, `ilab --config ./xxx` doesn't behave the same way as `ilab`, (though one could expect it to, I think.) Perhaps that's a bug in CLI definition, but I am not sure. Maybe @tiran will have an idea.

Reply:
You were on the right track! PR #1887 solves the problem in a generic way and also ensures that we do not introduce the same problem ever again.